AutomatticTracks/src/test/java/com/automattic/android/tracks/CrashLogging/CrashLoggingTest.kt

```diff
@@ -0,0 +1,70 @@
+package com.automattic.android.tracks.crashlogging
+
+import com.automattic.android.tracks.TracksUser
+import com.automattic.android.tracks.fakes.FakeDataProvider
+import com.automattic.android.tracks.fakes.testUser
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.never
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+class CrashLoggingTest {
+
+    lateinit var dataProvider: CrashLoggingDataProvider
+    var user: TracksUser? = null
+
+    private fun initialize(currentUser: TracksUser? = null) {
+
+        user = if (currentUser != null) spy(currentUser) else null
+
+        dataProvider = spy(FakeDataProvider(currentUser = user))
+
+        CrashLogging.start(
+            context = RuntimeEnvironment.systemContext,
+            dataProvider = dataProvider
+        )
+    }
```

Comment:
> because it leads to the possibility of the value being changed 

I'm not sure if I understand you correctly, could you please deliberate on that?

JUnit creates a separate class for every executed test so I don't see a possibility that tests will interfere with each other if you meant that.

I'm ok with refactoring this as you propose, it looks good too.